reana_workflow_controller/rest.py

```diff
@@ -428,10 +427,13 @@ def seed_workflow_workspace(workflow_id_or_name):
                                                    user_uuid)
 
         filename = full_file_name.split("/")[-1]
+
+        # Remove starting '/' in path
+        if full_file_name[0] == '/':
+            full_file_name = full_file_name[1:]
         path = get_analysis_files_dir(workflow, file_type,
                                       'seed')
-        if len(full_file_name.split("/")) > 1 and not \
-           os.path.isabs(full_file_name):
+        if len(full_file_name.split("/")) > 1:
```

Comment:
Indeed, I had tested manually and paths starting with '..' were failing, but I added the explicit check in: https://github.com/reanahub/reana-client/pull/79/commits/7c82db04b09d9df4c567e2012da9eda1c56cb7a7#diff-4f27f08c3f46f0b755c1d5ac1a1e2dcaR468
Maybe the warning could be more informative, but a better phrasing didn't come to me yet.

It looks like this now:
```
(reana-cwl-3.5) ➜  reana-demo-root6-roofit git:(master) ✗ reana-client code upload ../jq/autom4te.cache/ code
../jq/autom4te.cache/: Path must be under the upload type directory.
code/gendata.C was uploaded successfully.
code/code2 was uploaded successfully.
```

Reply:
It may be good to iterate over subdirs in `r-w-controller` anyway, and skip the ones equal to  `..` for additional safety.  We could simply emit an error to boot if we find `..` in path parts.  

This would add additional safety for the future, because there can be other clients besides `r-client` calling our REST API server one day.  Let's make the check on the server side as well then, inside `r-w-controller` for now, and later on perhaps in `r-server`. 

BTW the following user scenario still looks OK:

```console
$ cd ./docs/step17/
$ less some-instructions.txt
$ vim ../../code/step17/mycode.py
$ reana-client code upload ../../code/step17/mycode.py
```

> Path must be under the upload type directory.

It may be more understandable to say:  "path must be under the code directory" so that people understand the code vs inputs distinction.  (However beware of use case scenario where user uses only one directory for both code and inputs.)

Reply:
Just pushed in https://github.com/reanahub/reana-client/pull/79/files this check:

```
(reana-cwl-3.5) ➜  reana-demo-root6-roofit git:(master) ✗ reana-client code upload code/../../jq/AUTHORS 
code/../../jq/AUTHORS: Path cannot contain ".."
```
and removed the prepend of the upload type.

Reply:
@tiborsimko I just added the check for the paths including `../` here.